frontend/src/hyperglosae.js

```diff
@@ -20,17 +20,19 @@ function Hyperglosae(logger) {
       .then(x => x.json());
 
   this.putDocument = (doc, uri) =>
-    fetch(`${service}/${uri || doc._id}`, {
+    fetch(`${service}/_design/app/_update/document/${uri || doc._id}`, {
       method: 'PUT',
       body: JSON.stringify(doc)
     })
-      .then(x => x.json())
-      .then(x => {
-        if (x.reason) {
-          logger(x.reason);
-          throw new Error(x.reason);
+      .then(async x => {
+        const newRev = x.headers.get('X-Couch-Update-NewRev');
+        const response = await x.json();
+        if (response.reason) {
+          logger(response.reason);
+          throw new Error(response.reason);
         }
-        return x;
+        response.rev = newRev;
+        return response;
```

Comment:
Have you tried using `then` instead of `async/await` (for coding style coherence)?

Something like (please test and adapt):
```js

   .then(x => {
         const rev = x.headers.get('X-Couch-Update-NewRev');
         x.json().then(o => {
           if (o.reason) {
             logger(o.reason);
             throw new Error(o.reason);
         }
         return ({...o, rev});
   });
```